bundles/org.openhab.binding.vesync/src/main/java/org/openhab/binding/vesync/internal/handlers/VeSyncBridgeHandler.java

```diff
@@ -62,19 +69,31 @@ public class VeSyncBridgeHandler extends BaseBridgeHandler implements VeSyncClie
     private static final int DEFAULT_DEVICE_SCAN_RECOVERY_INTERVAL = 60;
     private static final int DEFAULT_DEVICE_SCAN_DISABLED = -1;
 
-    private final Logger logger = LoggerFactory.getLogger(VeSyncBridgeHandler.class);
-
-    private @Nullable ScheduledFuture<?> backgroundDiscoveryPollingJob;
+    private volatile int backgroundScanTime = -1;
 
     protected final VeSyncV2ApiHelper api = new VeSyncV2ApiHelper();
-    private IHttpClientProvider httpClientProvider;
-
-    private volatile int backgroundScanTime = -1;
+    private final Logger logger = LoggerFactory.getLogger(VeSyncBridgeHandler.class);
     private final Object scanConfigLock = new Object();
 
-    public VeSyncBridgeHandler(Bridge bridge, @NotNull IHttpClientProvider httpClientProvider) {
+    private final TranslationProvider translationProvider;
+    private final LocaleProvider localeProvider;
+    private final Bundle bundle;
+
+    private @Nullable ScheduledFuture<?> backgroundDiscoveryPollingJob;
+    private HttpClient httpClient;
+
+    public VeSyncBridgeHandler(Bridge bridge, @Reference HttpClientFactory httpClientFactory,
+            @Reference TranslationProvider translationProvider, @Reference LocaleProvider localeProvider) {
         super(bridge);
-        this.httpClientProvider = httpClientProvider;
+        this.httpClient = httpClientFactory.getCommonHttpClient();
```

Comment:
Why store the httpClient? Maybe better to pass the httpclient directly to the `VeSyncV2ApiHelper` ?

Reply:
Agreed 👍 - cleaned up - I went through quickly as a quick initial pass to remove the interface bits that were not required, from the old way of getting the client.